### ISSUE TITLE
Unexplained hang when github repo 404s

### ISSUE THREAD

**Issue by cthrax** [bower/bower#1967]
Hi All,

I think this may be an issue, feel free to close it if not, but I spent several hours beating my head against it.

TLDR:
If a github repo is missing, "bower install" (with or without "--verbose") gives no indication the repo is missing, but rather just hangs indefinitely. "bower list" appropriately produces the missing repo message.

Full Story and background:

I have a bower.json that I've been using for a few years now. I recently got a new computer, updated bower from 1.4.0 to 1.6.3 for as part of the new computer install. Went to run bower install and it hung after saying a particular dependency was resolved. Thinking maybe it was a version issue, I tried each version backwards to 1.4.0, with the same hang (though on different dependencies).

Accidentally, I typed "bower list" instead of "bower cache list" and lo and behold it tells me one of the github repos I have as a dependency no longer exists, I verified that and sure enough it did not exist. Removed the dependency and now "bower install" works fine. 

It would be really nice if "bower install" would indicate a missing repo.

**Comment by Utsav2** [comment]
I tried this on 1.6.5 on Arch Linux and it seems to work fine. Which OS are you using? Can you give any other details?

**Comment by cthrax** [comment]
I was running OSX yosemite, the dependency was in the bower directory, not a manual github url, I was using node version 4.2.1 and I'm not sure what else would be pertinent or useful information. Let me know and I'd be happy to get it.

**Comment by sheerun** [comment]
It's because bower is unuable to distinghush between repo that doesn't exist and repo that you don't have access to, so it asks for credentials. Running bower in non-interactive mode should help.
